frontend/src/views/WheresMyGene/components/GeneSearchBar/components/QuickSelect/index.tsx

```diff
@@ -253,9 +254,16 @@ export default function QuickSelect<
   };
   const selectedAsArray = Array.isArray(selected) ? selected : [selected];
 
-  const itemOptionsWithSelectedOnTop = [
-    ...new Set([...selectedAsArray, ...items]),
-  ];
+  const [itemOptions, setItemOptions] = useState<T[]>(items);
```

Comment:
It wouldn't. It needs to be stateful so that you don't get any visual glitch in the dropdown (e.g. temporary clearing of options) while the client is loading.